src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -877,14 +914,20 @@ private bool OpenFile(WorkspaceHeader workspaceInfo, XmlDocument xmlDoc, out Wor
                 nodeGraph.Notes,
                 workspaceInfo.X,
                 workspaceInfo.Y,
-                DebugSettings.VerboseLogging, IsTestMode, nodeGraph.ElementResolver, workspaceInfo.FileName);
+                DebugSettings.VerboseLogging, 
+                IsTestMode, 
+                nodeGraph.ElementResolver, 
+                workspaceInfo.RunType, 
+                workspaceInfo.RunPeriod,
+                workspaceInfo.FileName
```

Comment:
Dear lord, we've got to use a better pattern for initializing workspaces.  This is too much.  Why not pass in the whole `WorkspaceHeader` object?